spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -258,6 +295,125 @@ protected abstract MessageProducer createConsumerEndpoint(ConsumerDestination de
+		}
+		String recovererBeanName = getErrorRecovererName(destination, group, extendedConsumerProperties);
+		beanFactory.registerSingleton(recovererBeanName, recoverer);
+		beanFactory.initializeBean(recoverer, recovererBeanName);
+		MessageHandler handler = getErrorMessageHandler(destination, group, extendedConsumerProperties);
+		if (handler == null) {
+			handler = getDefaultErrorMessageHandler(errorChannel);
+		}
+		String errorMessageHandlerName = getErrorMessageHandlerName(destination, group, extendedConsumerProperties);
+		beanFactory.registerSingleton(errorMessageHandlerName, handler);
+		beanFactory.initializeBean(handler, errorMessageHandlerName);
+		errorChannel.subscribe(handler);
+		return new ErrorInfrastructure(errorChannel, recoverer, handler);
+	}
+
+	/**
+	 * Binders can return a message handler to be subscribed to the error channel.
+	 * Examples might be if the user wishes to (re)publish messages to a DLQ.
+	 * @param destination the destination.
+	 * @param group the group.
+	 * @param extendedConsumerProperties the properties.
+	 * @return the handler (may be null, which is the default, causing the exception to be
+	 * rethrown).
+	 */
+	protected MessageHandler getErrorMessageHandler(final ConsumerDestination destination, final String group,
+			final ExtendedConsumerProperties<?> extendedConsumerProperties) {
+		return null;
+	}
+
+	/**
+	 * Return the default error message handler, which throws the error message payload to
+	 * the caller if there are no user handlers subscribed. The handler is ordered so it
+	 * runs after any user-defined handlers that are subscribed.
+	 * @param errorChannel the error channel.
+	 * @return the handler.
+	 */
+	protected MessageHandler getDefaultErrorMessageHandler(final SubscribableChannel errorChannel) {
+		class FinalRethrowingErrorMessageHandler implements MessageHandler, Orderable {
+
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				if (errorChannel instanceof SubscriberAwareChannel) {
+					if (((SubscriberAwareChannel) errorChannel).subscribers() > 1) {
+						// user has subscribed
+						return;
+					}
+				}
+				if (message.getPayload() instanceof MessagingException) {
+					throw (MessagingException) message.getPayload();
+				}
+				else {
+					throw new MessagingException((Message<?>) null, (Throwable) message.getPayload());
+				}
+			}
+
+			@Override
+			public int getOrder() {
+				return Integer.MAX_VALUE; // run last if the user has subscribers too
+			}
+
+			@Override
+			public void setOrder(int order) {
+			}
+
+		}
+		return new FinalRethrowingErrorMessageHandler();
+	}
+
+	/**
+	 * Binders can return an {@link ErrorMessageStrategy} for building error messages; binder
+	 * implementations typically might add extra headers to the error message.
+	 * @return the implementation - may be null.
+	 */
+	protected ErrorMessageStrategy getErrorMessageStrategy() {
+		return null;
```

Comment:
That happens at the SI level - see this code in the binder...

		if (errorMessageStrategy == null) {
			recoverer = new ErrorMessageSendingRecoverer(errorChannel);
		}
		else {
			recoverer = new ErrorMessageSendingRecoverer(errorChannel, errorMessageStrategy);
		}